### DISCUSSION TITLE
/404 listed under Route (pages)

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#50718] [App Router]
I have a `not-found.tsx` that is listed under `app/(website)/not-found.tsx`. However, when building my app, it lists it under `Route (pages)`.

```
Route (app)                                Size     First Load JS
┌ ○ /                                      0 B                0 B
├ λ /[slug]                                0 B                0 B
├ λ /admin/[[...index]]                    0 B                0 B
├ λ /articles/[slug]                       0 B                0 B
├ λ /category/[category]                   0 B                0 B
├ ○ /robots.txt                            0 B                0 B
├ λ /search                                0 B                0 B
├ ○ /sitemap.xml                           0 B                0 B
└ ● /tag/[tag]                             0 B                0 B
    ├ /tag/nursing
    ├ /tag/vulnerable
    ├ /tag/age-friendly-arizona
    └ [+166 more paths]
+ First Load JS shared by all              77.7 kB
  ├ chunks/2443530c-26bd1c2cb2f545c8.js    50.5 kB
  ├ chunks/488-33323bca2ae3f1d3.js         24.9 kB
  ├ chunks/main-app-b7127ea921b99928.js    218 B
  └ chunks/webpack-d3903b3afdb1f74a.js     2.06 kB

Route (pages)                              Size     First Load JS
─ ○ /404                                   183 B          75.3 kB
+ First Load JS shared by all              75.1 kB
  ├ chunks/framework-43665103d101a22d.js   45.1 kB
  ├ chunks/main-03fab88100dd9b77.js        27.7 kB
  ├ chunks/pages/_app-b34b2dcd2ffe424c.js  198 B
  └ chunks/webpack-d3903b3afdb1f74a.js     2.06 kB
```

Any ideas why that would be?

**Comment by devjiwonchoi** [comment]


**Reply by JamesSingleton** [reply]
So are you saying that the `Route (pages)` is Next's global 404 page? I have nothing in `pages/` in my code.

**Reply by devjiwonchoi** [reply]
Apologies for the previous answer being incorrect. I've updated the answer.

**Comment by devjiwonchoi** [comment]
When using a [Route Groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups), you require a folder inside in order to direct the path for your simply saying, `member of the Route Groups`.

For example, `/(website)/something`, and the `not-found.js` as `/(website)/not-found.js`.

<img width="229" alt="Screenshot 2023-06-03 at 3 09 35 PM" src="https://github.com/vercel/next.js/assets/120007119/a82b5817-f904-479f-9b2a-cc668bf067e8">

By doing this, you can have the `not-found` file both on the `root` segment and inside the `(website)/something`.

Therefore any path to `/something/*` will show the `not-found.js` you are intending.

Since the `not-found.js` is technically not on the `root` segment, it is required to [call the `notFound` function](https://nextjs.org/docs/app/api-reference/functions/not-found) on your `Route Group` page.

> Note: In addition to catching expected notFound() errors, the root app/not-found.js file also handles any unmatched URLs for your whole application. This means users that visit a URL that is not handled by your app will be shown the UI exported by the app/not-found.js file.

Example:
```js
import { notFound } from 'next/navigation';

export default function Home() {
  // wrap the case for notFound(). For now, I've simply called it in order to test it.
  notFound();
  return (
    <div>
      <h1>TEST</h1>
    </div>
  );
}
  // ...
}
```

I've tested it out and it works fine.

<img width="331" alt="Screenshot 2023-06-03 at 3 12 29 PM" src="https://github.com/vercel/next.js/assets/120007119/6a6b5fe8-96bc-465b-bfe4-fe3b9d5359b0">
<img width="303" alt="Screenshot 2023-06-03 at 3 12 36 PM" src="https://github.com/vercel/next.js/assets/120007119/558b0a4e-8d1a-40b5-b16e-4aff0ed203e9">


<img width="542" alt="Screenshot 2023-06-03 at 3 13 50 PM" src="https://github.com/vercel/next.js/assets/120007119/f3d450f0-3a42-400b-ab1b-528393c86641">

## Reference
- App Router `notFound`: https://nextjs.org/docs/app/api-reference/functions/not-found

**Reply by jintak0401** [reply]
I have a question about the reason of `Route (pages) /404`. 

Whether I use Route Groups or not, I develop my app with app rotuer. So I think `Route (pages) /404` shouldn't be in build result because there is no `/pages` folder. The code for 404 page is at `/app/not-found.tsx` not `/pages/404.tsx`. Let me know why `Route (pages)` still be there.

**Reply by devjiwonchoi** [reply]
#48227 It's a bug not fixed yet. If you're able to generate `_not-found.html`, then no problem at this moment. Also, I guess it is there for pages that does not reach the `notFound` Error boundary as default.

**Comment by icyJoseph** [comment]
This is just the old 404 page that is still made. This has been also observed, incorrectly, as a source of larger JS bundles. 

I read from the Next.js team that they'll eventually do some check, that if you have `not-found.tsx` at top level in app dir, then they won't output the 404, but at this time the benefits of that are negligible.

> Because if your app is using app dir, then the pages/404 bundle is never shipped to the client, unless you don't have a `not-found.tsx`, when a 404 is needed.

- https://github.com/vercel/next.js/issues/49796#issuecomment-1547628161
